### PR TITLE
[MODULAR] Adds some examine detail to tallborgs

### PR DESCRIPTION
**Pull request by desertmote**
## About The Pull Request
Gives examine flavor-text to the Meka cyborgs, they are now labeled with their appropriate company branding.

## How This Contributes To The Skyrat Roleplay Experience

Avenue for roleplay and pretty.
dreamseeker_nnGYlzy7eP

## Changelog
:cl:
add: Added some examine flavor to tallborgs
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/altborgs/code/tallborg/robot.dm [hunk 1]
```diff
@@ -18,6 +18,53 @@
...
+	. = ..()
+
+	var/mob/living/silicon/robot/robot = src
+	. += span_notice("<i>You examine [src] closer, and note the following...</i>")
+
+	switch(user.icon)
+		if(CYBORG_ICON_PEACEKEEPER_TALL)
+			. += "Its waist has <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> stamped onto it."
+		if(CYBORG_ICON_MINING_TALL)
+			. += "Its waist has <i>[span_cyan("Interdyne Pharmaceuticals")]</i> laser-cut onto it."
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/altborgs/code/tallborg/robot.dm]: Not sure about this one, sounds a bit counter-intuitive that they'd get cyborg models, of all things, from Syndicate contractors.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
On top of that, my main question is...

_Why would a pharmaceutical company produce cyborg shells?_

Like, you'd figure they'd have a sub-division for that, like, `Zeng-Hu Electronics` or `Zeng-Hu Robotics`.

**Comment by desertmote** [comment]
> On top of that, my main question is...
> 
> _Why would a pharmaceutical company produce cyborg shells?_
> 
> Like, you'd figure they'd have a sub-division for that, like, `Zeng-Hu Electronics` or `Zeng-Hu Robotics`.

In my mind Meka has always been produced by zeng hu pharmaceuticals because of their IPC chassis, i also tried to imitate the gun's branding

LnOIwdOgTK

> Not sure about this one, sounds a bit counter-intuitive that they'd get cyborg models, of all things, from Syndicate contractors.

i found it cool to give the mining meka to interdyne, the model looks syndicate and their facility is mining themed in the game. the thought of them having stolen schematics from zeng hu because theyre bad like that is cool too.

File: modular_skyrat/modules/altborgs/code/tallborg/robot.dm (hunk 1 above)
Comment (ghost) [reply modular_skyrat/modules/altborgs/code/tallborg/robot.dm]: Well, technically, they arent _strictly_ syndicate, and they already can have dart weapons sold by them via guncargo.
But, yeah, having CYBORG models sold by them feels off, especially considering that NT is supposed to be dominant in the AI/Cyborg industry IIRC

File: modular_skyrat/modules/altborgs/code/tallborg/robot.dm [hunk 2]
```diff
@@ -18,6 +18,53 @@
...
+				if("mekaserve_alt")
+					. += "Its skirt has a <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> label."
+				else
+					return
+		if(CYBORG_ICON_JANI_TALL)
+			. += "Its skirt has a <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> label."
+		if(CYBORG_ICON_SYNDIE_TALL)
+			. += "Its label is crudely scratched off..."
+	//	if(CYBORG_ICON_CLOWN_TALL)
+	//		. += ""
```
Comment (tf-4) [inline modular_skyrat/modules/altborgs/code/tallborg/robot.dm:61]: ?

File: modular_skyrat/modules/altborgs/code/tallborg/robot.dm [hunk 3]
```diff
@@ -18,6 +18,53 @@
...
+
+/mob/living/silicon/robot/examine_more(mob/user)
+	. = ..()
+
+	var/mob/living/silicon/robot/robot = src
+	. += span_notice("<i>You examine [src] closer, and note the following...</i>")
+
+	switch(user.icon)
+		if(CYBORG_ICON_PEACEKEEPER_TALL)
+			. += "Its waist has <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> stamped onto it."
```
Comment (tf-4) [inline modular_skyrat/modules/altborgs/code/tallborg/robot.dm]:
```suggestion
			. += "It has <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> stamped onto its waist."
```
Just slightly awkward wording here imo

File: modular_skyrat/modules/altborgs/code/tallborg/robot.dm (hunk 1 above)
Comment (tf-4) [inline modular_skyrat/modules/altborgs/code/tallborg/robot.dm]:
```suggestion
			. += "It has <i>[span_cyan("Interdyne Pharmaceuticals")]</i> laser-cut into its waist."
```

File: modular_skyrat/modules/altborgs/code/tallborg/robot.dm [hunk 4]
```diff
@@ -18,6 +18,53 @@
...
+		if(CYBORG_ICON_MINING_TALL)
+			. += "Its waist has <i>[span_cyan("Interdyne Pharmaceuticals")]</i> laser-cut onto it."
+		if(CYBORG_ICON_MED_TALL)
+			. += "Its chest has <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> stamped onto it."
```
Comment (tf-4) [inline modular_skyrat/modules/altborgs/code/tallborg/robot.dm]:
```suggestion
			. += "It has <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> stamped onto its chest."
```

File: modular_skyrat/modules/altborgs/code/tallborg/robot.dm [hunk 5]
```diff
@@ -18,6 +18,53 @@
...
+		if(CYBORG_ICON_CARGO_TALL)
+			. += "Its coat has a <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> label."
+	//	if(CYBORG_ICON_ENG_TALL)
+	//		. += ""
+		if(CYBORG_ICON_SERVICE_TALL)
+			switch(robot.model.cyborg_base_icon)
+				if("mekaserve")
+					. += "Its fore-arm has <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> stamped onto it."
```
Comment (tf-4) [inline modular_skyrat/modules/altborgs/code/tallborg/robot.dm]:
```suggestion
					. += "It has <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> stamped onto its forearm."
```

File: modular_skyrat/modules/altborgs/code/tallborg/robot.dm [hunk 6]
```diff
@@ -18,6 +18,53 @@
...
+
+	switch(user.icon)
+		if(CYBORG_ICON_PEACEKEEPER_TALL)
+			. += "Its waist has <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> stamped onto it."
+		if(CYBORG_ICON_MINING_TALL)
+			. += "Its waist has <i>[span_cyan("Interdyne Pharmaceuticals")]</i> laser-cut onto it."
+		if(CYBORG_ICON_MED_TALL)
+			. += "Its chest has <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> stamped onto it."
+		if(CYBORG_ICON_CARGO_TALL)
+			. += "Its coat has a <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> label."
```
Comment (tf-4) [inline modular_skyrat/modules/altborgs/code/tallborg/robot.dm]:
```suggestion
			. += "It has a <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> label on its coat."
```

File: modular_skyrat/modules/altborgs/code/tallborg/robot.dm [hunk 7]
```diff
@@ -18,6 +18,53 @@
...
+	//	if(CYBORG_ICON_ENG_TALL)
+	//		. += ""
+		if(CYBORG_ICON_SERVICE_TALL)
+			switch(robot.model.cyborg_base_icon)
+				if("mekaserve")
+					. += "Its fore-arm has <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> stamped onto it."
+				if("mekaserve_alt")
+					. += "Its skirt has a <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> label."
```
Comment (tf-4) [inline modular_skyrat/modules/altborgs/code/tallborg/robot.dm]:
```suggestion
					. += "It has a <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> label on its skirt."
```

File: modular_skyrat/modules/altborgs/code/tallborg/robot.dm [hunk 8]
```diff
@@ -18,6 +18,53 @@
...
+				else
+					return
+		if(CYBORG_ICON_JANI_TALL)
+			. += "Its skirt has a <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> label."
```
Comment (tf-4) [inline modular_skyrat/modules/altborgs/code/tallborg/robot.dm]:
```suggestion
			. += "It has a <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> label on its skirt."
```

File: modular_skyrat/modules/altborgs/code/tallborg/robot.dm [hunk 9]
```diff
@@ -18,6 +18,53 @@
...
+		if(CYBORG_ICON_SYNDIE_TALL)
+			. += "Its label is crudely scratched off..."
+	//	if(CYBORG_ICON_CLOWN_TALL)
+	//		. += ""
+		if(CYBORG_ICON_NINJA_TALL)
+			. += "Its back has a <i>[span_green("Spider Clan")]</i> logo etched onto it."
```
Comment (tf-4) [inline modular_skyrat/modules/altborgs/code/tallborg/robot.dm]:
```suggestion
			. += "It has a <i>[span_green("Spider Clan")]</i> logo etched into its back."
```

File: modular_skyrat/modules/altborgs/code/tallborg/robot.dm (hunk 2 above)
Comment (SarmentiCampbell) [reply modular_skyrat/modules/altborgs/code/tallborg/robot.dm:61]: clown model coming up soon problably

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
Not really a fan of this PR for a few reasons
1) It's done in a really snowflake way that makes it a pain to easily add more in the future
2) The fact that it's only being done for tallborgs, I'd rather have it be done with all (at least, most) borgs or none
It reads to me as a very "made for me" PR

**Comment by desertmote** [comment]
what about this makes it a pain? its just a switch list? i made it amost 1:1 to how gunbrand are. also yeah its only for tallborgs because thats my OC. A ton of cyborg things ive done so far was just for tallborgs. why is it a problem now? the file its written in is just for tallborg code

**Comment by desertmote** [comment]
i want to ground Meka into the universe more and ill end up continueing to do so, because i want to keep some control about this entity i made. 

these cyborg models are just not Nanotrasen, the regular ones are, and the positronic/mmi tech is. **also i will not touch wideborgs,** those can stay brandless and unmaintained for all i care, sorry.

File: modular_skyrat/modules/altborgs/code/tallborg/robot.dm [hunk 10]
```diff
@@ -18,6 +18,53 @@
 	else
 		regenerate_icons()
 
+// Brand stamps for cyborgs
+/mob/living/silicon/robot/examine(mob/user)
+	. = ..()
+
+	var/mob/living/silicon/robot/robot = src
+	if(R_TRAIT_TALL in robot.model.model_features)
```
Comment (Zonespace27) [inline modular_skyrat/modules/altborgs/code/tallborg/robot.dm:26]: this is the snowflakey-ness that i'm talking about, it doesn't rely on a flag or anything, just a trait (exclusive to tallborgs)

File: modular_skyrat/modules/altborgs/code/tallborg/robot.dm [hunk 11]
```diff
@@ -18,6 +18,53 @@
...
+		. += span_notice("<i>It seems to have branding insignia, you can examine again to take a closer look...")
+	else
+		return
```
Comment (Zonespace27) [inline modular_skyrat/modules/altborgs/code/tallborg/robot.dm:29]: don't need this

File: modular_skyrat/modules/altborgs/code/tallborg/robot.dm [hunk 12]
```diff
@@ -18,6 +18,53 @@
...
+	//	if(CYBORG_ICON_ENG_TALL)
+	//		. += ""
+		if(CYBORG_ICON_SERVICE_TALL)
+			switch(robot.model.cyborg_base_icon)
+				if("mekaserve")
+					. += "It has <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> stamped onto its forearm."
+				if("mekaserve_alt")
+					. += "It has a <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> label on its skirt."
+				else
+					return
```
Comment (Zonespace27) [inline modular_skyrat/modules/altborgs/code/tallborg/robot.dm:55]: ditto

File: modular_skyrat/modules/altborgs/code/tallborg/robot.dm [hunk 13]
```diff
@@ -18,6 +18,53 @@
...
+			. += "It has a <i>[span_pink("Zeng-Hu Pharmaceuticals")]</i> label on its skirt."
+		if(CYBORG_ICON_SYNDIE_TALL)
+			. += "Its label is crudely scratched off..."
+	//	if(CYBORG_ICON_CLOWN_TALL)
+	//		. += ""
+		if(CYBORG_ICON_NINJA_TALL)
+			. += "It has a <i>[span_green("Spider Clan")]</i> logo etched into its back."
+
+		else
+			return
```
Comment (Zonespace27) [inline modular_skyrat/modules/altborgs/code/tallborg/robot.dm:66]: ditto

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Comment by Zonespace27** [comment]
You haven't touched reviews and I'm still not interested in this. Feel free to try and get this reopened if you convince a maintainer it's good

**Pull request closed without merging**